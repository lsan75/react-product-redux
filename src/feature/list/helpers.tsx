import { IProduct } from '../../store/product/product'

export const productList: IProduct[] = [
  {
    name: 'aaa',
    price: 10,
    selected: true
  },
  {
    name: 'bbb',
    price: 20,
    selected: false
  },
  {
    name: 'ccc',
    price: 50
  }
]