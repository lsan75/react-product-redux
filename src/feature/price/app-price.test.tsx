import * as React from 'react'
import * as createMockStore from 'redux-mock-store'

import * as Adapter from 'enzyme-adapter-react-16'

import { configure, mount, ReactWrapper, shallow, ShallowRendererProps, ShallowWrapper } from 'enzyme'

import { mountToJson, shallowToJson } from 'enzyme-to-json'

import { Provider } from 'react-redux'
import thunk from 'redux-thunk'

import ConnectedContainer, { IProps } from './app-price'

import { productList } from '../../mocks/helpers'

describe('AppList', () => {
  configure({ adapter: new Adapter() })

  // @ts-ignore
  const mockStore = createMockStore([ thunk ])
  const store = mockStore({
    productReducer: {
      productList,
      totalPrice: 10
    }
  })
  const props: ShallowRendererProps = {
    context: { store }
  }
  let shallowOutput: ShallowWrapper<IProps>
  let reactOutput: ReactWrapper<IProps>


  beforeEach(() => {
    reactOutput = mount<IProps>(
      <Provider store={store}>
        <ConnectedContainer />
      </Provider>
    )
    shallowOutput = shallow<IProps>(
      <Provider store={store}>
        <ConnectedContainer />
      </Provider>
    ).dive(props)
  })

  afterEach(() => {
    store.clearActions()
    shallowOutput.unmount()
    reactOutput.unmount()
  })

  it('renders without crashing', () => {
    const innerProps: IProps = shallowOutput.props()

    expect(shallowToJson(shallowOutput)).toMatchSnapshot()
    expect(mountToJson(reactOutput)).toMatchSnapshot()
    expect(innerProps.productList).toEqual([{
      name: 'aaa',
      price: 10,
      selected: true
    }])
    expect(innerProps.totalPrice).toBe(10)
  })

})
