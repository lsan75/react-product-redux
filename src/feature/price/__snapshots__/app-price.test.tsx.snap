// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppList renders without crashing 1`] = `
<AppPrice
  dispatch={[Function]}
  productList={
    Array [
      Object {
        "name": "aaa",
        "price": 10,
        "selected": true,
      },
    ]
  }
  totalPrice={10}
/>
`;

exports[`AppList renders without crashing 2`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AppPrice)>
    <AppPrice
      dispatch={[Function]}
      productList={
        Array [
          Object {
            "name": "aaa",
            "price": 10,
            "selected": true,
          },
        ]
      }
      totalPrice={10}
    >
      <div
        className="Wrapper"
      >
        <PriceHeader
          totalPrice={10}
        >
          <article
            className="TotalPrice"
          >
            <h2
              className="TotalPrice__Title"
            >
              Total
            </h2>
            <span
              className="TotalPrice__Value"
            >
              10
               €
            </span>
          </article>
        </PriceHeader>
        <PriceList
          products={
            Array [
              Object {
                "name": "aaa",
                "price": 10,
                "selected": true,
              },
            ]
          }
        >
          <ul
            className="PriceList"
          >
            <li
              className="PriceList__Item"
              key="aaa"
            >
              <strong
                className="PriceList__Item__Name"
              >
                aaa
              </strong>
              <span
                className="PriceList__Item__Price"
              >
                10
                 €
              </span>
            </li>
          </ul>
        </PriceList>
      </div>
    </AppPrice>
  </Connect(AppPrice)>
</Provider>
`;
